src/interactive/scala/tools/nsc/interactive/Global.scala

```diff
@@ -301,6 +301,7 @@ class Global(settings: Settings, _reporter: Reporter, projectName: String = "")
   def isOutOfDate: Boolean = outOfDate
 
   def demandNewCompilerRun() = {
+    if (!lastWasReload) allSources.foreach(getUnit(_).foreach(reset(_)))
```

Comment:
would it be equivalent to add a boolean param to `demandNewCompilerRun` and set it true at the callsite in `reload`?